chainstate/src/detail/chainstateref.rs

```diff
@@ -682,6 +703,16 @@ impl<'a, S: BlockchainStorageWrite, O: OrphanBlocksMut> ChainstateRef<'a, S, O>
             prev_block_index.block_height().next_height()
         });
 
+        let some_ancestor = match prev_block_index {
+            Some(ref prev_bi) => Some(
+                self.get_ancestor(prev_bi, get_skip_height(height))
+                    .map_err(|_| BlockError::AncestorRetrievalFailed(block.get_id()))?
+                    .block_id()
+                    .clone(),
```

Comment:
Could you explain why is the peer banned if ancestor retrieval fails? I'm not sure if I fully understand this change because to me it seems we're calculating the height based on the data we have locally available (on line 702) but when the skip height is calculated and that ancestor is not found, it's a peer error. It sounds more like a database inconsistency to me so what am I missing?

Reply:
Yep... it's a database consistency or an implementation issue. You're right. But this is, again, the issue of fearing having done something wrong and causing an attack vector.

Reply:
After rethinking this, I changed it to an expect and added more aggressive tests to ensure all corner cases are covered. Let's hope I missed nothing.